src/extension.ts

```diff
@@ -16,20 +17,30 @@ import { KeyChain } from './credential/keyChain';
 import { DocsEnvironmentController } from './common/docsEnvironmentController';
 import { BuildStatusBarObserver } from './observers/buildStatusBarObserver';
 import { CodeActionProvider } from './codeAction/codeActionProvider';
+import { ExtensionContext } from './extensionContext';
+import config from './config';
+import { EnvironmentController } from './common/environmentController';
```

Comment:
why do we need the interface `EnvironmentController`, if only `DocsEnvironmentController` implements it?

Reply:
For test usage, since the implementation of DocsEnvironmentController depends on the vscode API, we can easly moc it after extacting an interface.

---
In reply to: [379341655](https://github.com/docascode/vscode-docs-build/pull/41#discussion_r379341655) [](ancestors = 379341655)